astropy/uncertainty/core.py

```diff
@@ -220,6 +220,21 @@ def pdf_percentiles(self, percentile, **kwargs):
             return self.distribution._new_view(percs)
         else:
             return percs
+            
+    def pdf_asymstd(self):
+        """ 
+        a convenience function that computes the 50% percentile
+        value and the asymmetric uncertainties around that value.
+        ( p50, p84-p50, p50-p16).
+        
+        Returns
+        -------
+        percentiles : tuple of three `~astropy.units.Quantity`
+                   50% percentile value and the asymmetric uncertainties
+                   around that value.( p50, p84-p50, p50-p16).           
+        """
+        p50 = self.pdf_percentiles(50)
```

Comment:
I wonder if it would be faster to call `pdf_percentiles` with all three needed values and then using the stored values for the `return` statement.  This would avoid calling this function 3 times.

Reply:
yeah youre right (considering numpy code is wriiten in such a way),iam gonna write a code as per the changes you have  suggested.

Reply:
i will choose your algo and write the codes(i thought about doing that way when i realised when we input a list it the nppercent function output as an np array so i thought that the np function have to deal with list function  such as pop(i) and wasnt sure nparraray index[] fuction is o(1) like list i  said this if we put one inout there are many list cancellation happening ,but after thinking about it i think the new way is bettter .)